### PR TITLE
Report aria-label and aria-description even when parent has no visible children

### LINKED ISSUE #14514: A <figure>'s aria-label and description is not spoken when it has a child element with aria-hidden="true"
### Steps to reproduce:
Given the following html:
```
<body>
  <h1>Heading 1</h1>
  <h2>Heading 2</h2>
  <figure aria-label="Label for figure" aria-description="A Description">
    <p aria-hidden="true">I'm paragraph text</p>
  </figure>
</body>
```

NVDA will not read the label or description for the figure element. This appears to be due to the child `<p>` having aria-hidden="true". When the aria-hidden attribute is removed NVDA will read the figure's label and description as expected.

Also, changing the `<figure>` to a `<div>` with a role="figure" reads as expected, so the following will read the label and description while properly ignoring the child `<p>`:
```
<body>
  <h1>Heading 1</h1>
  <h2>Heading 2</h2>
  <div aria-role="figure" aria-label="Label for figure" aria-description="A Description">
    <p aria-hidden="true">I'm paragraph text</p>
  </div>
</body>
```

### Actual behavior:
The `<figure>` label and description attributes are not read when there is a child element that has aria-hidden="true"

### Expected behavior:
NVDA should read the label and description attributes and ignore the hidden child elements using the `<figure>` element.

#### NVDA version:
2022.4.0.27401

#### Windows version:
Version	10.0.19044 Build 19044

#### Name and version of other software in use when reproducing the issue:
Tested in Firefox and Chrome

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

### PR DESCRIPTION
**Pull request by SaschaCowley**
### Link to issue number:

Fixes #14514 

### Summary of the issue:

HTML `figure` elements with no accessible children (EG, all children set to `aria-hidden=true`) but with an `aria-label` or `aria-description` are not reported by NVDA.

### Description of user facing changes

Elements with no accessible children but which have a label or description are now reported in browse mode. Note this only applies to elements which are still rendered; if all children are set to `display: none`, for example, NVDA still does not report the parent element.

### Description of development approach

Render a single space in elements with no content but a label or description set.

### Testing strategy:

Manually tested in Firefox and Edge, using the [example markup provided by the OP](https://github.com/nvaccess/nvda/issues/14514#issue-1521143638).

### Known issues with pull request:

N/A

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by seanbudd** [comment]
is this ready for review?

File: nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp [hunk 1]
```diff
@@ -1262,11 +1262,13 @@ VBufStorage_fieldNode_t* GeckoVBufBackend_t::fillVBuf(
...
 		}
 
-		if ((isInteractive || role == ROLE_SYSTEM_SEPARATOR) && parentNode->getLength() == 0) {
+		if ((isInteractive || role == ROLE_SYSTEM_SEPARATOR || name || description.has_value()) && parentNode->getLength() == 0) {
 			// If the node is interactive or otherwise relevant even when empty
 			// and it still has no content, render a space so the user can access the node.
 			previousNode = buffer->addTextFieldNode(parentNode, previousNode, EMPTY_TEXT_NODE);
-			if(previousNode&&!locale.empty()) previousNode->addAttribute(L"language",locale);
+			if(previousNode&&!locale.empty()) {
+				previousNode->addAttribute(L"language",locale);
```
Comment (seanbudd) [inline nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp:1270]: Lint fix

```suggestion
			if (previousNode&&!locale.empty()) {
				previousNode->addAttribute(L"language", locale);
```

**Review by seanbudd — APPROVED** [review]

**Comment by seanbudd** [comment]
Could you move the changes to the new changes.md for this PR please?

**Comment by LeonarddeR** [comment]
I'm afraid this pr has unexpected side effects. Se https://github.com/nvaccess/nvda/issues/16554

File: nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp [hunk 2]
```diff
@@ -1262,11 +1262,13 @@ VBufStorage_fieldNode_t* GeckoVBufBackend_t::fillVBuf(
 			parentNode->isBlock=false;
 		}
 
-		if ((isInteractive || role == ROLE_SYSTEM_SEPARATOR) && parentNode->getLength() == 0) {
+		if ((isInteractive || role == ROLE_SYSTEM_SEPARATOR || name || description.has_value()) && parentNode->getLength() == 0) {
```
Comment (LeonarddeR) [inline nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp:1265]: Should `name` be changed to `name.has_value()` here?

Comment (SaschaCowley) [reply nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp:1265]: Not exactly, as `name` is a `BSTR`, not an `optional<wstring>`. I have checked that it's non-null and non-empty in my updated code though.

**Pull request merged**